solr/core/build.gradle

```diff
@@ -90,6 +90,7 @@ dependencies {
   implementation 'org.apache.httpcomponents:httpcore'
 
   compileOnlyApi 'javax.servlet:javax.servlet-api'
+  compileOnlyApi 'org.eclipse.jetty.toolchain:jetty-servlet-api'
```

Comment:
> I don't think we need this 

I tried without it and it failed with a client/server using SSL mismatch before putting this iteration up. I can take one more look at it.

> I think having both of these is incorrect? 

I never actually added that myself, it did exist when I started, it was removed in a rebase I did, and then, and if I remove it now, analyzeClassesDependencies fails for me with (I'm digging around a bit, not sure why it existed, or why it could be removed but then not in this issue):


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':solr:modules:jwt-auth:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - org.eclipse.jetty.toolchain:jetty-servlet-api:4.0.6@jar


Execution failed for task ':solr:modules:jwt-auth:analyzeClassesDependencies'.
> Dependency analysis found issues.
  usedUndeclaredArtifacts
   - org.eclipse.jetty.toolchain:jetty-servlet-api:4.0.6@jar

Reply:
Not sure what you want to do here. If I remove that, analyzeClassDependencies will fail.

Here is the details on that toolchain module, appears to be part of what is needed to build Jetty: https://github.com/eclipse/jetty.toolchain

Here is a dependency insight call on that lib:

org.eclipse.jetty.toolchain:jetty-servlet-api:4.0.6
+--- project :
|    \--- runtimeClasspath
+--- org.eclipse.jetty:jetty-rewrite:10.0.8
|    +--- runtimeClasspath
|    \--- project : (requested org.eclipse.jetty:jetty-rewrite:{strictly 10.0.8}) (*)
\--- org.eclipse.jetty:jetty-server:10.0.8
     +--- runtimeClasspath
     +--- project : (requested org.eclipse.jetty:jetty-server:{strictly 10.0.8}) (*)
     +--- io.dropwizard.metrics:metrics-jetty9:4.1.5 (requested org.eclipse.jetty:jetty-server:9.4.27.v20200227)
     |    +--- runtimeClasspath
     |    \--- project : (requested io.dropwizard.metrics:metrics-jetty9:{strictly 4.1.5}) (*)
     +--- org.eclipse.jetty:jetty-rewrite:10.0.8 (*)
     +--- org.eclipse.jetty:jetty-security:10.0.8
     |    +--- runtimeClasspath
     |    +--- project : (requested org.eclipse.jetty:jetty-security:{strictly 10.0.8}) (*)
     |    \--- org.eclipse.jetty:jetty-servlet:10.0.8
     |         +--- runtimeClasspath
     |         +--- project : (requested org.eclipse.jetty:jetty-servlet:{strictly 10.0.8}) (*)
     |         \--- org.eclipse.jetty:jetty-webapp:10.0.8
     |              +--- runtimeClasspath
     |              +--- project : (requested org.eclipse.jetty:jetty-webapp:{strictly 10.0.8}) (*)
     |              \--- org.eclipse.jetty:jetty-deploy:10.0.8
     |                   +--- runtimeClasspath
     |                   \--- project : (requested org.eclipse.jetty:jetty-deploy:{strictly 10.0.8}) (*)
     +--- org.eclipse.jetty:jetty-alpn-server:10.0.8
     |    +--- runtimeClasspath
     |    +--- project : (requested org.eclipse.jetty:jetty-alpn-server:{strictly 10.0.8}) (*)
     |    \--- org.eclipse.jetty:jetty-alpn-java-server:10.0.8
     |         +--- runtimeClasspath
     |         \--- project : (requested org.eclipse.jetty:jetty-alpn-java-server:{strictly 10.0.8}) (*)
     \--- org.eclipse.jetty.http2:http2-server:10.0.8
          +--- runtimeClasspath
          \--- project : (requested org.eclipse.j

Reply:
This is addressed now